x-pack/plugins/ml/server/models/calculate_model_memory_limit/calculate_model_memory_limit.js

```diff
@@ -105,7 +105,7 @@ export function calculateModelMemoryLimitProvider(callWithRequest) {
           const maxBytes = numeral(limits.max_model_memory_limit.toUpperCase()).value();
           const mmlBytes = numeral(`${mmlMB}MB`).value();
           if (mmlBytes > maxBytes) {
-            mmlMB = (maxBytes / numeral('1MB').value());
+            mmlMB = Math.ceil(maxBytes / numeral('1MB').value());
```

Comment:
Rounding up would generally violate the limit.  I think you'll often get away with it because you are using pedantic `KB` and `MB`.

I think if you want to be pedantic you should use `KiB` and `MiB`.  Everyone else does in the world of computers except the marketing departments of hard disk manufacturers.  ES is using 1 MB = 1024 * 1024 bytes.

There is a further complication in the formula `maxBytes = numeral(limits.max_model_memory_limit.toUpperCase()).value()` where an `i` needs to be inserted before any `B` to convert `MB` to `MiB`, `KB` to `KiB`, etc.

Reply:
we're using a forked version of `numeral` where ``` _numeral.default(`1MB`).value();``` = 1048576, so this shouldn't be a problem.
However, it appears numeral is rounding up when converting fractions. e.g. `2.7GB` is `2899102925` in numeral, rather than the exact value of `2899102924.8`
So to be safe I'll change this `Math.ceil` to a `Math.floor` to ensure we never go over the max mml due to a rounding error